net/WebSocketHandler.hpp

```diff
@@ -671,11 +675,15 @@ class WebSocketHandler : public ProtocolHandlerInterface
         if (_pingTimeout.count() > std::numeric_limits<double>::epsilon() &&
             _pingMicroS.average() >= _pingTimeout.count())
         {
-            LOG_WRN("CheckTimeout: Timeout websocket: Ping: last " << _pingMicroS.last() << "us, avg "
-                    << _pingMicroS.average() << "us >= " << _pingTimeout.count() << "us over "
-                    << (int)_pingMicroS.duration() << "s");
-            shutdownSilent();
-            return true;
+            std::shared_ptr<StreamSocket> socket = _socket.lock();
```

Comment:
Why the lock and the copy when this comes in as a parameter ?

Reply:
I'm quite disturbed by closing connections on longer average pings too - where does that come from ?

Reply:
> I'm quite disturbed by closing connections on longer average pings too - where does that come from ?

I described the source (and idea) of WS ping TO instrument https://github.com/CollaboraOnline/online/pull/10334#issuecomment-2439712761 as well as wrapped it up in #10337 

I am currently about to finish my draft PR for #10337  and will post its initial version,
currently refining the UTs for it.

Reply:
> Why the lock and the copy when this comes in as a parameter ?

the ` _socket.lock()` has moved up to `checkTimeout` to cover for the `socket->isIPType()` check, which you have correctly introduced with your change request (but not coded all out).